app/controllers/deployments_controller.rb

```diff
@@ -48,10 +56,29 @@ def create
 private
 
   def application_by_repo
-    if (existing_app = Application.find_by(repo: repo_path))
-      existing_app
+    existing_apps = Application.where(repo: repo_path)
+    if existing_apps.present?
+      if existing_apps.length == 1
+        existing_apps[0]
+      else
+        flash[:alert] = format("Found multiple applications using repo: %<repo_path>s while using application_by_repo",
+                               repo_path: repo_path)
+        render :new
+        return nil
+      end
     else
-      Application.create!(name: app_name, repo: repo_path, domain: domain)
+      Application.create!(name: normalize_app_name(repo_path), repo: repo_path, domain: domain)
+    end
+  end
+
+  def application_by_name
+    existing_apps = Application.where(repo: repo_path, name: params[:application_name])
+
+    case existing_apps.length
+    when 1
+      existing_apps[0]
+    when 0
+      Application.create!(name: normalize_app_name(params[:application_name]), repo: repo_path, domain: domain)
```

Comment:
Would it be possible to switch the `case` statement from here to `application_by_repo`.

There are 3 outcomes in `application_by_repo`, and I think having a flat `case` statement would be more readable than a nested `if`. Also, as there are only two outcomes in `application_by_name`, an `if` statement would be better there I think.

Reply:
I see you've done this now, I think it does look better :+1: